src/utils.rs

```diff
@@ -667,3 +667,74 @@ pub fn smallest_abs_diff_between_angles<N: SimdRealCopy>(a: N, b: N) -> N {
     let s_err_is_smallest = s_err.simd_abs().simd_lt(s_err_complement.simd_abs());
     s_err.select(s_err_is_smallest, s_err_complement)
 }
+
+/// Helpers around serialization.
+#[cfg(feature = "serde-serialize")]
+pub mod serde {
+    use serde::{Deserialize, Serialize};
+    use std::iter::FromIterator;
+
+    /// Serializes to a `Vec<(K, V)>`.
+    ///
+    /// Useful for [`std::collections::HashMap`] with a non-string key,
+    /// which is unsupported by [`serde_json`].
+    pub fn serialize_to_vec_tuple<
+        'a,
+        S: serde::Serializer,
+        T: IntoIterator<Item = (&'a K, &'a V)>,
+        K: Serialize + 'a,
+        V: Serialize + 'a,
+    >(
+        target: T,
+        s: S,
+    ) -> Result<S::Ok, S::Error> {
+        let container: Vec<_> = target.into_iter().collect();
+        serde::Serialize::serialize(&container, s)
+    }
+
+    /// Deserializes from a `Vec<(K, V)>`.
+    ///
+    /// Useful for [`std::collections::HashMap`] with a non-string key,
+    /// which is unsupported by [`serde_json`].
+    #[cfg(feature = "serde-serialize")]
```

Comment:
Not needed since the `pub mod serde` is already feature-gated.

Reply:
I was conflicted between gating the whole mod or each functions ; gating the whole mod might give users a bit of a pain for their "use" statements, but it's not a huge annoyance, and it feels cleaner to gate the whole mod